src/next/components/Snippet/UsercentricsSmartDataProtector.tsx

```diff
@@ -0,0 +1,53 @@
+"use client";
+
+import React, {ReactElement} from "react";
+import Script from "next/script";
+
+type WindowWithUsercentricsSmartDataProtector = Window & Readonly<{
+	uc?: {
+		blockElements(elements: Record<string, string>): void;
+		reloadOnOptIn(key: string): void;
+		reloadOnOptOut(key: string): void;
+	};
+}>;
+
+export const USERCENTRICS_SERVICE_PROVIDER_KEYS = {
+	GOOGLE_MAPS: "S1pcEj_jZX",
+} as const;
+
+type UsercentricsSerivceProviderInternalKeys = keyof typeof USERCENTRICS_SERVICE_PROVIDER_KEYS;
+type UsercentricsSerivceProviderKey = typeof USERCENTRICS_SERVICE_PROVIDER_KEYS[UsercentricsSerivceProviderInternalKeys] | string;
+type HTMLSelector = string;
+
+type UsercentricsSmartDataProtectorProps = Readonly<{
+	elements?: Record<UsercentricsSerivceProviderKey, HTMLSelector>;
+}>;
+
+export function UsercentricsSmartDataProtector (props: UsercentricsSmartDataProtectorProps): ReactElement | null
+{
+	const keys = props.elements ? Object.keys(props.elements) : [];
```

Comment:
IMO we shouldn't include this script if no services configuration is passed. Or does it make sense to still include it for its automatic logic, which may still be useful?

Reply:
It makes sense to include this anyway, as a YouTube, for example, can be recognized automatically. With `elements` you only specify special cases that are not in an iframe on the page, e.g. Google Maps in a div. I have added a description to the prop, which also mentions this, as well as `reloadKeys`, where the non-special cases can also be added to `reloadOnOptIn` and `reloadOnOptOut`.